data/render/src/lib.rs

```diff
@@ -21,6 +23,65 @@ pub fn validate(tags: &Tags, entries: &Vec<Entry>) -> Result<(), Box<dyn Error>>
     Ok(())
 }
 
+#[tokio::main]
+pub async fn check_deprecated(token: std::string::String, entries: &mut Vec<Entry>) -> Result<Vec<Entry>, Box<dyn Error>> {
+    let github = Github::new(
+        String::from("user-agent-name"),
+        Credentials::Token(token),
+    )?;
+
+    let mut entries_tmp: Vec<Entry> = entries.to_vec();
+    for entry in &mut entries_tmp {
+        if entry.source.is_none() {
+            continue;
+        }
+
+        let mut source: &str = entry.source.as_ref().unwrap();
+        if source.chars().last().unwrap() == '/' {
+            source = source.trim_end_matches('/');
+        }
+
+        let components: Vec<&str> = source.split("/").collect();
+        if !(components.contains(&"github.com") && components.len() == 5) {
+            // valid github source must have 5 elements - anything longer and they are probably a
+            // reference to a path inside a repo, rather than a repo itself.
+            continue;
+        }
+
+        let owner = components[3];
+        let repo  = components[4];
+
+        let commit_list = github
+            .repo(owner, repo)
+            .commits()
+            .list()
+            .await;
+
+        let commit_list = match commit_list {
+            Ok(commit_list) => commit_list,
+            Err(_error) => Vec::new(),
+        };
+        if commit_list.len() == 0 {
+            continue;
+        }
+
+        let date: &str = &commit_list[0].commit.author.date[..];
+        let timestamp = NaiveDateTime::parse_from_str(date, "%Y-%m-%dT%H:%M:%SZ")?.timestamp();
+        let current_timestamp = Utc::now().timestamp();
+
+        if current_timestamp - timestamp > 365 * 86400 {
```

Comment:
`chrono` has ways to deal with durations, I just can't give you a good example.
It's *something* like that, but not quite...

```rust
    let last_commit = NaiveDateTime::parse_from_str(&date, "%Y-%m-%dT%H:%M:%SZ")?;
    let today = Local::today();

    let duration = today.signed_duration_since(last_commit);
    
    println!("Today is: {}", today.format("%Y-%m-%d"));
```

Reply:
maybe something like this?

```rust
let date = &commit_list[0].commit.author.date;
let last_commit = NaiveDateTime::parse_from_str(&date, "%Y-%m-%dT%H:%M:%SZ")?;
let last_commit_utc = DateTime::<Utc>::from_utc(last_commit, Utc);
let duration = Local::today().signed_duration_since(last_commit_utc.date());

if duration.num_days() > 365 {
    entry.deprecated = Some(true);
} else {
    entry.deprecated = None;
}
```

Reply:
Yes, that's what I meant. Perfect!